modules/nlu/src/backend/typings.ts

```diff
@@ -36,6 +36,14 @@ export interface KnownSlot extends sdk.NLU.SlotDefinition {
   source: string
 }
 
+export interface SlotValidation extends KnownSlot {
+  isValidEntity: boolean
+}
+
+export interface IntentValidation {
```

Comment:
Here we validate only slots. Do you think we'll eventually add other types of validation for each utterances ? If no then you might want to call this as IntentSlotValidation.

Otherwise,. maybe we would like to add an object for each utterance. 

```
IntentValidation {
  [utterance:string]: {
    slots : SlotValidation[],
    hasDupes : boolean,
    /* anything else ?*/
  }
}
```